### PR TITLE
[Validate] Refactor GF

### PR DESCRIPTION
**Pull request by israelpoli**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-10052

### CODE REVIEW

File: demisto_sdk/commands/validate/tests/GF_validators_test.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        - Ensure that the ValidationResult returned
+          for the GenericField whose 'group' field is not valid
+        - Ensure that no ValidationResult returned when group field set to 4
+    """
+    # not valid
+    generic_field = create_generic_field_object(paths=["group"], values=[0])
+    assert GenericFieldGroupValidator().is_valid([generic_field])
+
+    # valid
+    generic_field.group = 4
```
Comment (dorschw) [inline demisto_sdk/commands/validate/tests/GF_validators_test.py:55]:
```suggestion
    generic_field.group = REQUIRED_GROUP_VALUE
```

File: demisto_sdk/commands/validate/tests/GF_validators_test.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    When:
+        - run fix method
+    Then:
+        - Ensure the fix message as expected
+        - Ensure the field `id` changing to with `generic_` prefix
+    """
+
+    generic_field = create_generic_field_object(paths=["id"], values=["foo"])
+
+    result = GenericFieldIdPrefixValidateValidator().fix(generic_field)  # type:ignore
```
Comment (dorschw) [inline demisto_sdk/commands/validate/tests/GF_validators_test.py:88]: why `type:ignore`?

File: demisto_sdk/commands/validate/validators/GF_validators/GF100_generic_field_group.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+REQUIRED_GROUP_VALUE = 4
+ContentTypes = GenericField
+
+
+class GenericFieldGroupValidator(BaseValidator[ContentTypes]):
+    error_code = "GF100"
+    rationale = "Required by the platform."
+    description = f"Checks if group field is set to {REQUIRED_GROUP_VALUE}."
+    error_message = "The `group` key must be set to 4 for Generic Field"
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GF_validators/GF100_generic_field_group.py:20]:
```suggestion
    error_message = f"The `group` key must be set to {REQUIRED_GROUP_VALUE}"
```

File: demisto_sdk/commands/validate/validators/GF_validators/GF100_generic_field_group.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    fix_message = f"`group` field is set to {REQUIRED_GROUP_VALUE}."
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GF_validators/GF100_generic_field_group.py:21]:
```suggestion
    fix_message = f"set the`group` field to {REQUIRED_GROUP_VALUE}."
```

File: demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+GENERIC_FIELD_ID_PREFIX = "generic_"
+ContentTypes = GenericField
+
+
+class GenericFieldIdPrefixValidateValidator(BaseValidator[ContentTypes]):
+    error_code = "GF101"
+    rationale = "Required by the platform."
+    description = "Checks if `id` field include prefix `generic_`"
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py:19]:
```suggestion
    description = "Checks if the id starts with `generic_`"
```

File: demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    error_message = "ID `{generic_id}` is not valid, it should start with the prefix `{generic_id_prefix}`."
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py:20]:
```suggestion
    error_message = "{generic_id} is not a valid id, it should start with {generic_id_prefix}."
```

File: demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    fix_message = "Change the value of `id` field to `{generic_id}`."
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py:21]:
```suggestion
    fix_message = "Changed the value of `id` field to `{generic_id_prefix}_{generic_id}`."
```
right?

**Review by dorschw — COMMENTED** [review]
Good job! see the few comments.

File: demisto_sdk/commands/content_graph/parsers/generic_field.py [hunk 8]
```diff
@@ -18,7 +18,7 @@ def __init__(
         super().__init__(path, pack_marketplaces, git_sha=git_sha)
         self.definition_id = self.json_data.get("definitionId")
         self.field_type = self.json_data.get("type") or ""
-
+        self.group = self.json_data.get("group")
```
Comment (YuvHayun) [inline demisto_sdk/commands/content_graph/parsers/generic_field.py:21]: Let's add it as a property.
see cli_name in parsers/incident_field.py as a reference (make sure to add it to the field_mapping dict as well.

File: demisto_sdk/commands/validate/validators/GF_validators/GF100_generic_field_group.py (hunk 4 above)
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/GF_validators/GF100_generic_field_group.py:21]:
```suggestion
    description = f"Validate that the 'group' field is set to {REQUIRED_GROUP_VALUE}."
    error_message = f"The 'group' key must be set to {REQUIRED_GROUP_VALUE}."
    fix_message = f"Change the 'group' field value to {REQUIRED_GROUP_VALUE}."
```

File: demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py (hunk 7 above)
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py:21]:
```suggestion
    description = "Validate that the 'id' field starts with 'generic_'"
    error_message = "The content item ID ({generic_id}) is invalid, it muist start with the prefix '{generic_id_prefix}'."
```

File: demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    related_field = "id"
+    is_auto_fixable = True
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py:23]:
```suggestion
```
Let's remove the fix here, we don't want to take the risk of modifying calls in the customer's env here.

File: demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+            if (not content_item.object_id.startswith(GENERIC_FIELD_ID_PREFIX))
+        ]
+
+    def fix(self, content_item: ContentTypes) -> FixResult:
+        content_item.object_id = GENERIC_FIELD_ID_PREFIX + content_item.object_id
+        return FixResult(
+            validator=self,
+            message=self.fix_message.format(generic_id=content_item.object_id),
+            content_object=content_item,
+        )
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py:45]:
```suggestion
```

File: demisto_sdk/commands/validate/tests/GF_validators_test.py [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    assert result
+    assert result.message == "Change the value of `id` field to `generic_foo`."
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/GF_validators_test.py:90]:
```suggestion
```

File: demisto_sdk/commands/validate/tests/GF_validators_test.py (hunk 2 above)
Comment (israelpoli) [reply demisto_sdk/commands/validate/tests/GF_validators_test.py:88]: my mistake

**Review by YuvHayun — APPROVED** [review]
Achla IP!
Make sure to change the tests according to the changes in the validations.

File: demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py (hunk 7 above)
Comment (israelpoli) [reply demisto_sdk/commands/validate/validators/GF_validators/GF101_generic_field_id_prefix_validate.py]: @YuvHayun asked to download the fix option
so deleted completely

**Review by dorschw — APPROVED** [review]

**Pull request merged**